google/cloud/internal/curl_wrappers.cc

```diff
@@ -243,26 +255,60 @@ std::size_t CurlAppendHeaderData(CurlReceivedHeaders& received_headers,
   return size;
 }
 
+std::string DebugInfo(char const* data, std::size_t size) {
+  return absl::StrCat("== curl(Info): ", absl::string_view{data, size});
+}
+
+std::string DebugRecvHeader(char const* data, std::size_t size) {
+  return absl::StrCat("<< curl(Recv Header): ", absl::string_view{data, size});
+}
+
+std::string DebugSendHeader(char const* data, std::size_t size) {
+  auto header = absl::string_view{data, size};
+  auto const match = absl::string_view{": Bearer "};
+  auto const limit = match.size() + 32;
+  auto const bearer_pos = header.find(match);
+  if (bearer_pos != std::string::npos) {
+    auto const nl_pos = header.find('\n', bearer_pos);
+    auto const n = (std::min)(limit, nl_pos - bearer_pos);
+    return absl::StrCat(
+        ">> curl(Send Header): ", header.substr(0, bearer_pos + n),
+        n == limit ? "...<truncated>..." : "",
```

Comment:
>> Nothing is truncated if `n` == `nl_pos - bearer_pos`, right?
> 
> Fixed

You chose to test `n == limit`, which then mistakenly announces truncation when `limit == nl_pos - bearer_pos`.

Reply:
Sorry about that, I believe it is fixed now.